File: WoofIOS/SwiftPackages/Data/DataSource/Sources/DataSource/Dogs/Fixtures/DogBreeds.swift
import DTO
import Foundation

struct DogBreed {
    let name: String
    let photoSlug: String
    let size: String
    let climates: Set<Climate>
}

enum DogBreeds {
    private static let anyClimate = Set(Climate.allCases)
    private static let coldClimate: Set<Climate> = [.cold]
    private static let coolClimate: Set<Climate> = [.cold, .temperate]
    private static let mildClimate: Set<Climate> = [.temperate, .warm]

    static let all: [DogBreed] = [
        DogBreed(name: "Golden Retriever", photoSlug: "golden_retriever", size: DogSizeDTO.large, climates: anyClimate),
        DogBreed(name: "Labrador Retriever", photoSlug: "labrador_retriever", size: DogSizeDTO.large, climates: mildClimate),
        DogBreed(name: "French Bulldog", photoSlug: "french_bulldog", size: DogSizeDTO.small, climates: mildClimate),
        DogBreed(name: "German Shepherd", photoSlug: "german_shepherd", size: DogSizeDTO.large, climates: anyClimate),
        DogBreed(name: "Poodle", photoSlug: "poodle", size: DogSizeDTO.medium, climates: anyClimate),
        DogBreed(name: "Beagle", photoSlug: "beagle", size: DogSizeDTO.medium, climates: anyClimate),
        DogBreed(name: "Dachshund", photoSlug: "dachshund", size: DogSizeDTO.small, climates: anyClimate),
        DogBreed(name: "Corgi", photoSlug: "corgi", size: DogSizeDTO.small, climates: anyClimate),
        DogBreed(name: "Border Collie", photoSlug: "border_collie", size: DogSizeDTO.medium, climates: anyClimate),
        DogBreed(name: "Siberian Husky", photoSlug: "siberian_husky", size: DogSizeDTO.large, climates: coldClimate),
        DogBreed(name: "Boxer", photoSlug: "boxer", size: DogSizeDTO.large, climates: anyClimate),
        DogBreed(name: "Great Dane", photoSlug: "great_dane", size: DogSizeDTO.large, climates: anyClimate),
        DogBreed(name: "Shiba Inu", photoSlug: "shiba_inu", size: DogSizeDTO.medium, climates: anyClimate),
        DogBreed(name: "Australian Shepherd", photoSlug: "australian_shepherd", size: DogSizeDTO.medium, climates: coolClimate),
        DogBreed(name: "Bernese Mountain Dog", photoSlug: "bernese_mountain_dog", size: DogSizeDTO.large, climates: coldClimate),
        DogBreed(name: "Jack Russell Terrier", photoSlug: "jack_russell_terrier", size: DogSizeDTO.small, climates: anyClimate),
        DogBreed(name: "Pug", photoSlug: "pug", size: DogSizeDTO.small, climates: mildClimate),
        DogBreed(name: "Samoyed", photoSlug: "samoyed", size: DogSizeDTO.large, climates: coldClimate),
        DogBreed(name: "Dalmatian", photoSlug: "dalmatian", size: DogSizeDTO.large, climates: anyClimate),
        DogBreed(name: "Chihuahua", photoSlug: "chihuahua", size: DogSizeDTO.small, climates: mildClimate),
    ]

    static func named(_ name: String) -> DogBreed {
        all.first { $0.name == name }!
    }
}

File: WoofAndroid/data/datasource/src/main/kotlin/com/perrystreet/woof/datasource/dogs/fixtures/DogBreeds.kt
package com.perrystreet.woof.datasource.dogs.fixtures

import com.perrystreet.woof.dto.dog.DogSizeDTO

internal data class DogBreed(
    val name: String,
    val photoSlug: String,
    val size: String,
    val climates: Set<Climate>,
)

internal object DogBreeds {
    private val AnyClimate = Climate.entries.toSet()
    private val ColdClimate = setOf(Climate.Cold)
    private val CoolClimate = setOf(Climate.Cold, Climate.Temperate)
    private val MildClimate = setOf(Climate.Temperate, Climate.Warm)

    val all: List<DogBreed> = listOf(
        DogBreed("Golden Retriever", "golden_retriever", DogSizeDTO.Large, AnyClimate),
        DogBreed("Labrador Retriever", "labrador_retriever", DogSizeDTO.Large, MildClimate),
        DogBreed("French Bulldog", "french_bulldog", DogSizeDTO.Small, MildClimate),
        DogBreed("German Shepherd", "german_shepherd", DogSizeDTO.Large, AnyClimate),
        DogBreed("Poodle", "poodle", DogSizeDTO.Medium, AnyClimate),
        DogBreed("Beagle", "beagle", DogSizeDTO.Medium, AnyClimate),
        DogBreed("Dachshund", "dachshund", DogSizeDTO.Small, AnyClimate),
        DogBreed("Corgi", "corgi", DogSizeDTO.Small, AnyClimate),
        DogBreed("Border Collie", "border_collie", DogSizeDTO.Medium, AnyClimate),
        DogBreed("Siberian Husky", "siberian_husky", DogSizeDTO.Large, ColdClimate),
        DogBreed("Boxer", "boxer", DogSizeDTO.Large, AnyClimate),
        DogBreed("Great Dane", "great_dane", DogSizeDTO.Large, AnyClimate),
        DogBreed("Shiba Inu", "shiba_inu", DogSizeDTO.Medium, AnyClimate),
        DogBreed("Australian Shepherd", "australian_shepherd", DogSizeDTO.Medium, CoolClimate),
        DogBreed("Bernese Mountain Dog", "bernese_mountain_dog", DogSizeDTO.Large, ColdClimate),
        DogBreed("Jack Russell Terrier", "jack_russell_terrier", DogSizeDTO.Small, AnyClimate),
        DogBreed("Pug", "pug", DogSizeDTO.Small, MildClimate),
        DogBreed("Samoyed", "samoyed", DogSizeDTO.Large, ColdClimate),
        DogBreed("Dalmatian", "dalmatian", DogSizeDTO.Large, AnyClimate),
        DogBreed("Chihuahua", "chihuahua", DogSizeDTO.Small, MildClimate),
    )

    fun named(name: String): DogBreed = all.first { it.name == name }
}
